drivers/adc/adc_smartbond_gpadc.c

```diff
@@ -101,6 +112,30 @@ static int adc_smartbond_channel_setup(const struct device *dev,
 	return 0;
 }
 
+static inline void gpadc_smartbond_pm_policy_state_lock_get(struct adc_smartbond_data *data)
+{
+#if defined(CONFIG_PM_DEVICE)
+	if (atomic_test_and_set_bit(data->pm_policy_state_flag, 0) == 0) {
```

Comment:
The API should return a boolean value. As such, comparison with integers should be avoided.

Reply:
Fixed